fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAutoCompleteResultCursor.java

```
@@ -169,12 +170,325 @@ private void performLookup() throws IOException {
+                if (rec == null) {
+                    // No document found. Return original record.
+                    return null;
+                }
+                // Extract the indexed fields from the document again
+                final List<LuceneDocumentFromRecord.DocumentField> documentFields = LuceneDocumentFromRecord.getRecordFields(state.index.getRootExpression(), rec)
+                        .get(groupingKey == null ? TupleHelpers.EMPTY : groupingKey);
+
+                // Search each field to find the first match.
+                final int maxTextLength = Objects.requireNonNull(state.context.getPropertyStorage()
+                        .getPropertyValue(LuceneRecordContextProperties.LUCENE_AUTO_COMPLETE_TEXT_SIZE_UPPER_LIMIT));
+                @Nullable LuceneDocumentFromRecord.DocumentField matchedField = null;
+                @Nullable String matchedText = null;
+                for (LuceneDocumentFromRecord.DocumentField documentField : documentFields) {
+                    Object fieldValue = documentField.getValue();
+                    if (fieldValue instanceof String) {
+                        String text = (String) fieldValue;
+                        if (text.length() > maxTextLength) {
+                            // Apply the text length filter before searching through the text for the
+                            // matched terms
+                            continue;
+                        }
+                        String match = searchAllMaybeHighlight(text, queryTokens, prefixToken, highlight);
+                        if (match != null) {
+                            matchedField = documentField;
+                            matchedText = match;
+                            break;
+                        }
+                    }
+                }
+
+                if (matchedField == null) {
+                    return null;
```

It might make more sense to throw exception here rather than returning ```null```. We need to understand why no matching fields are found from the matched doc.